File: app/src/main/java/com/example/challenge_consult_cep/feature/search_cep/domain/use_case/SearchCepUseCase.java
package com.example.challenge_consult_cep.feature.search_cep.domain.use_case;

import com.example.challenge_consult_cep.feature.search_cep.domain.model.Cep;
import com.example.challenge_consult_cep.feature.search_cep.domain.repository.SearchCepRepository;

import io.reactivex.rxjava3.core.Single;

public class SearchCepUseCase {

    private final SearchCepRepository repository;

    public SearchCepUseCase(
            SearchCepRepository repository
    ) {
        this.repository = repository;
    }

    public Single<Cep> invoke(String cep) {
        return repository.getCep(cep);
    }
}

File: app/src/main/java/com/example/challenge_consult_cep/feature/search_cep/domain/use_case/SearchCepUseCase.kt
package com.example.challenge_consult_cep.feature.search_cep.domain.use_case

import com.example.challenge_consult_cep.feature.search_cep.domain.model.Cep
import com.example.challenge_consult_cep.feature.search_cep.domain.repository.SearchCepRepository

class SearchCepUseCase(
    private val repository: SearchCepRepository
) {

    suspend fun getUsers(cep: String): Cep = repository.getCep(cep)
}